File: leetcode/1_Two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict = {}
        for i in range(0, len(nums)):
            if target - nums[i] in dict:
                return [dict[target - nums[i]], i]
            else:
                dict[nums[i]] = i

# second solution with get() and enumerate
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dic = {}

        for i, num in enumerate(nums):
            dic.get(i)
            compliment = target - num
            if compliment in dic:
                return [dic[compliment], i]
            else:
                dic[num] = i
        

File: leetcode/1_two_sum.cpp
using namespace std;
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> m;

        for (int i = 0; i < nums.size(); i++) {
            int need = target - nums[i];

            if (m.find(need) != m.end()) {
                return {m[need], i};
            } else {
                m[nums[i]] = i;
            }
        }
        return {};
    }
};
